esp/services/ws_packageprocess/ws_packageprocessService.cpp

```diff
@@ -681,11 +749,15 @@ bool CWsPackageProcessEx::readPackageMapString(const char *packageMapString, Str
     return true;
 }
 
-void CWsPackageProcessEx::getPkgInfoById(const char *packageMapId, IPropertyTree* tree)
+void CWsPackageProcessEx::getPkgInfoById(const char *target, const char *packageMapId, IPropertyTree* tree)
 {
     if (!packageMapId || !*packageMapId)
         return;
 
+    StringBuffer scopedPMID;
+    if (target && *target)
+        packageMapId = scopedPMID.append(target).append("::").append(packageMapId).str();
```

Comment:
With simple strings like this I error on the side of being more efficient... but that's because when its just something like "::" I personally find appends just as easy to read.  More complex hard to read strings I use appendf.